File: src/main/java/com/saybetter/domain/solution/domain/SolutionSymbol.java
package com.saybetter.domain.solution.domain;

import com.saybetter.domain.symbol.domain.Symbol;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Getter
@Entity(name = "SOLUTION_SYMBOL")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class SolutionSymbol {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "ss_id")
	private Long ssId;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "solution_id", nullable = false)
	private Solution solution;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "symbol_id", nullable = false)
	private Symbol symbol;

	@Builder
	public SolutionSymbol(Solution solution, Symbol symbol) {
		this.solution = solution;
		this.symbol = symbol;
	}

}

File: storage/storage-mysql-kt/src/main/kotlin/io/say/better/storage/mysql/domain/entity/SolutionSymbol.kt
package io.say.better.storage.mysql.domain.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity(name = "SOLUTION_SYMBOL")
class SolutionSymbol(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ss_id")
    private val ssId: Long = 0,
    @JoinColumn(name = "solution_id", nullable = false)
    @ManyToOne(fetch = FetchType.LAZY)
    private val solution: Solution,
    @JoinColumn(name = "symbol_id", nullable = false)
    @ManyToOne(fetch = FetchType.LAZY)
    private val symbol: Symbol,
)
